From f51b9de4d1a5ccc6fa786fe555dead6edf74eae8 Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Mon, 26 Feb 2018 02:07:05 +0530
Subject: [PATCH 0300/1851] drm: xlnx: csc: Add YUV420 support

commit 6202a885e6842fa7148f11456275b3f510c5d082 from
https://github.com/Xilinx/linux-xlnx.git

This commit adds the YUV 420 media bus format to color space converter driver.

Signed-off-by: Venkateshwar Rao G <venkateshwar.rao.gannavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_csc.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_csc.c b/drivers/gpu/drm/xlnx/xlnx_csc.c
index dad53e86a00c..824ab8fa0ba9 100644
--- a/drivers/gpu/drm/xlnx/xlnx_csc.c
+++ b/drivers/gpu/drm/xlnx/xlnx_csc.c
@@ -61,6 +61,7 @@ static const u32 xilinx_csc_video_fmts[] = {
 	MEDIA_BUS_FMT_RGB888_1X24,
 	MEDIA_BUS_FMT_VUY8_1X24,
 	MEDIA_BUS_FMT_UYVY8_1X16,
+	MEDIA_BUS_FMT_VYYUYY8_1X24,
 };
 
 /* vpss_csc_color_fmt - Color format type */
@@ -304,6 +305,9 @@ static int xilinx_csc_bridge_set_input(struct xlnx_bridge *bridge, u32 width,
 	case MEDIA_BUS_FMT_UYVY8_1X16:
 		csc->cft_in = XVIDC_CSF_YCRCB_422;
 		break;
+	case MEDIA_BUS_FMT_VYYUYY8_1X24:
+		csc->cft_in = XVIDC_CSF_YCRCB_420;
+		break;
 	default:
 		dev_dbg(csc->dev, "unsupported input video format\n");
 		return -EINVAL;
@@ -370,6 +374,12 @@ static int xilinx_csc_bridge_set_output(struct xlnx_bridge *bridge, u32 width,
 		if (csc->cft_in == MEDIA_BUS_FMT_RBG888_1X24)
 			xcsc_rgb_to_ycrcb(csc, &csc->clip_max);
 		break;
+	case MEDIA_BUS_FMT_VYYUYY8_1X24:
+		csc->cft_out = XVIDC_CSF_YCRCB_420;
+		dev_dbg(csc->dev, "Media Format Out : YUV 420");
+		if (csc->cft_in == MEDIA_BUS_FMT_RBG888_1X24)
+			xcsc_rgb_to_ycrcb(csc, &csc->clip_max);
+		break;
 	default:
 		dev_info(csc->dev, "unsupported output video format\n");
 		return -EINVAL;
-- 
2.31.1

